test/e2e/pages/page.js

```diff
@@ -50,7 +50,29 @@ export default class Page {
   async chooseLocation() {
     const count = await this.nodes.locationsList.count
     const randomItem = Math.floor(Math.random() * count)
-    return t.click(this.nodes.locationsList.nth(randomItem))
+
+    await t.expect(count).notEql(0)
+    await t.expect(this.getCurrentUrl()).contains('/locations')
```

Comment:
I am nitpicking here, but maybe check the url first and the dom rendering after? (e.g. switch the two statements).

Reply:
What @barbaracassani said